wp-components-php/inc/classes/class-renderable.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Renderable class.
+ *
+ * @package WP_Component
+ */
+
+namespace WP_Component\PHP;
+
+/**
+ * Renderable.
+ */
+class Renderable {
+
+	/**
+	 * Component instance
+	 *
+	 * @var array
+	 */
+	public $component_instance;
+
+	/**
+	 * Should the partial be output (false) or returned (true).
+	 *
+	 * @var bool Default false.
+	 */
+	public $return = false;
+
+	/**
+	 * Slug of template to load
+	 *
+	 * @var bool Default false.
+	 */
+	public $template_slug = '';
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param \Wp_Component\Component $component_instance Instance of a component (or template) to render
+	 */
+	public function __construct( $component_instance, $return = false ) {
+		$this->component_instance = $component_instance;
+		$this->return = $return;
+	}
+
+	/**
+	 * Load the partial.
+	 *
+	 * @param string $template_slug Slug of template to load.
+	 */
+	public function render( $template_slug = 'index' ) {
+		$this->template_slug = $template_slug ?? $this->template_slug;
```

Comment:
Is this needed since we're passing in a default value for `$template_slug`? I guess if `null` were explicitly passed?